projects/publishing-api/docker-compose.yml

```diff
@@ -34,6 +34,7 @@ services:
   publishing-api-app: &publishing-api-app
     <<: *publishing-api
     depends_on:
+      - govuk-content-schemas-app
```

Comment:
I don't think we need these, as Content Schemas just need to be present, not started, which is what the Makefiles ensure.

Does that make sense?

Reply:
If I omit them, I get test failures:

```
Failures:

  1) Make dependencies mirrors docker-compose.yml for publishing-api
     Failure/Error:
       expect(compose_dependencies(project_name))
         .to match_array(makefile_dependencies(project_name))
     
       expected collection contained:  ["govuk-content-schemas"]
       actual collection contained:    []
       the missing elements were:      ["govuk-content-schemas"]
     # ./spec/integration/make/dependencies_spec.rb:6:in `block (3 levels) in <top (required)>'

Finished in 0.32347 seconds (files took 0.36969 seconds to load)
1081 examples, 1 failure
```

Any ideas?

Reply:
Ahh, I see. 

Now that `govuk-content-schemas` has an `-app` stack, [it's getting picked up by `makefile_dependencies`](https://github.com/alphagov/govuk-docker/blob/main/spec/integration/make/dependencies_spec.rb#L18). The expectation is that if an app needs another "app" in its Makefile, then it also needs the app to be running when it starts. I think the schemas are the one exception to this, because it's just the files that matter.

Optional: we could clarify a few of the names in that file:

- `compose_dependencies` => `app_dependencies_in_compose_file`
- `makefile_dependencies` => `app_dependencies_in_makefile`

Thinking about it that way, we can see that the schemas aren't an "app dependency" and we could tweak that method to exclude it as a special case.

Does that help?